store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/util/ConcatenateSetMergeFunction.java

```diff
@@ -1,53 +0,0 @@
-/*
- * Copyright 2022 Crown Copyright
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *     http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package uk.gov.gchq.gaffer.federatedstore.util;
-
-import com.google.common.collect.Sets;
-import org.apache.commons.lang3.builder.HashCodeBuilder;
-
-import uk.gov.gchq.koryphe.impl.function.IterableConcat;
-import uk.gov.gchq.koryphe.impl.function.ToSet;
-
-import java.util.Collections;
-import java.util.function.BiFunction;
-
-import static java.util.Objects.isNull;
-import static java.util.Objects.nonNull;
-
-public class ConcatenateSetMergeFunction implements BiFunction<Object, Iterable<Object>, Iterable<Object>> {
-
-    @Override
-    public Iterable<Object> apply(final Object update, final Iterable<Object> state) {
-        //When update=null, Then this STOPS ToSet returning an Iterable with a null in it.
-        final Iterable<Object> updateSafe = isNull(update) ? Collections.emptySet() : (Iterable<Object>) new ToSet().apply(update);
-        return isNull(state)
-                ? updateSafe
-                : new IterableConcat<>().apply(Sets.newHashSet(updateSafe, state));
```

Comment:
How do I get a concatenation of of type set ?
(distinct ?)

Reply:
If you need a Set return type, put a `ToSet` operation after the one that was merged with this.
It will always be inefficient with a lazy iterable though as you need to empty the iterable into a Set to make it distinct.